src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -310,12 +310,14 @@ public function getProductType(ProductId $productId): ProductType
             ));
         }
 
-        if (!empty($result['product_type'])) {
-            return new ProductType($result['product_type']);
+        // Older products that were created before product page v2, might have no type, so we determine it dynamically
+        // According Phpstan, "Strict comparison using === between mixed and '' will always evaluate to false." so we use == instead of ===
+        if (empty($result['product_type']) || ($result['product_type'] == ProductType::TYPE_UNDEFINED)) {
```

Comment:
I think you should annotate the `$result` variable instead of using `==`:
```php
/** @var array<string, string> $result */
$result = $this->connection->createQueryBuilder()
...
```

Reply:
Lol, in fact I was completely out of subject, the point was that `ProductType::TYPE_UNDEFINED = ''` (empty string) so `empty($result['product_type'])` already test this case :sweat_smile: